addons/mrp/static/src/widgets/mrp_should_consume.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module **/
+
+import { FloatField } from '@web/views/fields/float/float_field';
+import { registry } from '@web/core/registry';
+import { formatFloat } from '@web/views/fields/formatters';
+
+/**
+ * This widget is used to display alongside the total quantity to consume of a production order,
+ * the exact quantity that the worker should consume depending on the BoM. Ex:
+ * 2 components to make 1 finished product.
+ * The production order is created to make 5 finished product and the quantity producing is set to 3.
+ * The widget will be '3.000 / 5.000'.
+ */
+
+const { useRef, onPatched, onMounted } = owl;
+export class MrpShouldConsumeOwl extends FloatField {
+    setup() {
+        super.setup();
+        const { data, fields } = this.props.record;
+        this.shouldConsumeQty = formatFloat(data.should_consume_qty, {
+            ...fields.should_consume_qty,
+            ...this.nodeOptions,
+        });
+        this.displayShouldConsume = !["done", "draft", "cancel"].includes(data.state);
+        this.prefixSpanRef = useRef("prefixSpan");
+        this.wrapSpanRef = useRef("wrapSpan");
+        this.inputSpanRef = useRef("numpadDecimal");
+        onMounted(this._renderPrefix);
+        onPatched(this._renderPrefix);
+    }
+
+    _renderPrefix() {
+        const wrapClass = "o_should_consume ps-1";
+        if (this.displayShouldConsume && this.prefixSpanRef.el && this.wrapSpanRef.el) {
```

Comment:
I think you can only keep `displayShouldConsume` because if `displayShouldConsume` is true, the prefix span and the wrap span will always be rendered, no ?